### PR TITLE
urls: prefer `/archive/refs/tags` urls over `/archive`

### PR DESCRIPTION
**Pull request by chenrui333**
unblocks https://github.com/Homebrew/brew/pull/16126

### CODE REVIEW

File: Formula/b/boolector.rb [hunk 1]
```diff
@@ -26,13 +26,13 @@ class Boolector < Formula
...
+    url "https://github.com/arminbiere/lingeling/archive/refs/tags/7d5db72420b95ab356c98ca7f7a4681ed2c59c70.tar.gz"
     sha256 "cf04c8f5706c14f00dd66e4db529c48513a450cc0f195242d8d0762b415f4427"
   end
 
   # Use commit has from `contrib/setup-btor2tools.sh`
   resource "btor2tools" do
-    url "https://github.com/boolector/btor2tools/archive/1df768d75adfb13a8f922f5ffdd1d58e80cb1cc2.tar.gz"
+    url "https://github.com/boolector/btor2tools/archive/refs/tags/1df768d75adfb13a8f922f5ffdd1d58e80cb1cc2.tar.gz"
     sha256 "cee19843635a15ad599424a5ad098669938afed85f6d3341e5d661cf7cd5b261"
   end
```
Comment (Bo98) [inline Formula/b/boolector.rb:37]: These don't seem correct

File: Formula/c/caffe.rb [hunk 2]
```diff
@@ -35,7 +35,7 @@ class Caffe < Formula
   end
 
   resource "homebrew-test_model" do
-    url "https://github.com/nandahkrishna/CaffeMNIST/archive/2483b0ba9b04728041f7d75a3b3cf428cb8edb12.tar.gz"
+    url "https://github.com/nandahkrishna/CaffeMNIST/archive/refs/tags/2483b0ba9b04728041f7d75a3b3cf428cb8edb12.tar.gz"
     sha256 "2d4683899e9de0949eaf89daeb09167591c060db2187383639c34d7cb5f46b31"
   end
```
Comment (Bo98) [inline Formula/c/caffe.rb:40]: Ditto

File: Formula/d/dashing.rb [hunk 3]
```diff
@@ -24,7 +24,7 @@ class Dashing < Formula
   depends_on "go" => :build
 
   resource "redux_saga_docs_tarball" do
-    url "https://github.com/dmitrytut/redux-saga-docset/archive/7df9e3070934c0f4b92d66d2165312bf78ecd6a0.tar.gz"
+    url "https://github.com/dmitrytut/redux-saga-docset/archive/refs/tags/7df9e3070934c0f4b92d66d2165312bf78ecd6a0.tar.gz"
     sha256 "08e5cc1fc0776fd60492ae90961031b1419ea6ed02e2c2d9db2ede67d9d67852"
   end
```
Comment (Bo98) [inline Formula/d/dashing.rb:29]: Ditto

File: Formula/l/luajit.rb [hunk 4]
```diff
@@ -10,7 +10,7 @@ class Luajit < Formula
   # Get the latest commit with:
   #   `git ls-remote --heads https://github.com/LuaJIT/LuaJIT.git v2.1`
   # This is a rolling release model so take care not to ignore CI failures that may be regressions.
-  url "https://github.com/LuaJIT/LuaJIT/archive/656ecbcf8f669feb94e0d0ec4b4f59190bcd2e48.tar.gz"
+  url "https://github.com/LuaJIT/LuaJIT/archive/refs/tags/656ecbcf8f669feb94e0d0ec4b4f59190bcd2e48.tar.gz"
```
Comment (Bo98) [inline Formula/l/luajit.rb:13]: Ditto

**Comment by Bo98** [comment]
+ many more - I didn't check them all.

As I said in https://github.com/Homebrew/brew/pull/16126#issuecomment-1773569625, this is wrong for commit URLs and the regex needs updating to exclude them.

File: Formula/c/caffe.rb (hunk 2 above)
Comment (chenrui333) [reply Formula/c/caffe.rb:40]: yeah, let me fix the script

**Comment by chenrui333** [comment]
make sense

**Comment by chenrui333** [comment]
excluded all the commit sha tgz urls from the changeset (and updated the audit regex in the brew side)

**Comment by chenrui333** [comment]
- ran `brew fetch` for the changeset, no checksum mismatch found
- reverted some changes for repo_removed formulae (like `neovide` and `nethacked`)

**Comment by SMillerDev** [comment]
This really shouldn't all be in one commit.

**Comment by chenrui333** [comment]
> This really shouldn't all be in one commit.

yeah, I was thinking about to do something similar to https://github.com/Homebrew/homebrew-core/pull/139592

it is mostly style changes though, I can go either way.

**Comment by Bo98** [comment]
> yeah, I was thinking about to do something similar to #139592

Plan was to rewrite history after that PR and it was useful at that point in time to have a single point in Git history that denoted the change, but history rewriting never happened in the end.

I wouldn't use that PR as precedence - it was a bit of a special PR that required changing repository settings to merge and doesn't match how we typically do `brew style --fix` PRs.

**Comment by chenrui333** [comment]
ok, let me spread the commits then :)

**Comment by chenrui333** [comment]
ok, I will provide two options :) (see how folks prefer)

**Review by MikeMcQuaid — APPROVED** [review]

**Pull request merged**